### PR TITLE
fix(style): #3504 Show snippets above other UI elements

### LINKED ISSUE #3504: Favicon and context menu icon show through snippets
Also see https://bugzilla.mozilla.org/show_bug.cgi?id=1401254
screen shot 2017-09-16 at 6 59 45 am

It looks like z-index isn't necessary for either to show up correctly? @sarracini did you run into some situation that required it for favicons (bottom right)?

I suppose the context menu (top right) showing on top of snippets might be okay, but the context menu icon (top left) can end up just showing through by itself if it has (keyboard) focus or hover.

### PR DESCRIPTION
**Pull request by k88hudson**
Fix #3504

### CODE REVIEW

File: system-addon/content-src/activity-stream.scss [hunk 1]
```diff
@@ -110,6 +110,11 @@ a {
   }
 }
 
+// Make sure snippets show up above other UI elements
+#snippets-container {
```
Comment (sarracini) [inline system-addon/content-src/activity-stream.scss]: can you also remove the z-index on the icons themselves? under ```.default-icon```

**Comment by Mardak** [comment]
#2547 added a `z-index: 399` for the context menu icon https://github.com/mozilla/activity-stream/commit/d52b2424faea4417126d98aa076a0ecd3f0c6c3b#diff-7e4f8acf366adfb4ebaad85c3126b9e0R64

I guess that's less likely to happen than the favicons.

**Comment by sarracini** [comment]
Did we decide to keep the context menu button on top of the snippets container?

**Comment by Mardak** [comment]
Onboarding uses `z-index: 20998` ;) https://searchfox.org/mozilla-central/source/browser/extensions/onboarding/content/onboarding.css#494

Although a related issue is onboarding covers up the pref pane, which is `z-index: 12000`
https://github.com/mozilla/activity-stream/blob/ec89268f747d5b92ec75ba1da91d11407286ab63/system-addon/content-src/components/PreferencesPane/_PreferencesPane.scss#L23

**Comment by Mardak** [comment]
I believe it was decided that the menu icon would be beneath but menu itself over. With the PR it looks like this right now (notice the bottom one):
image

**Comment by sarracini** [comment]
Right, so let's also remove the ```z-index: 399``` on ```.context-menu-button``` @k88hudson and that should be good. So many z-indexes 😂... err z-indices?

**Comment by as-pine-proxy** [comment]
[Treeherder: 5ce1edcfeac38561bd2ae69f1149ff3e6e0d3040](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=5ce1edcfeac38561bd2ae69f1149ff3e6e0d3040)

**Pull request merged**